Certificate of analysis 2.0.0

jacewski-bosch opened the pull request:
## Description
Certificate of Analysis has been added two optional properties Document Type and Disclaimer. This way it contains all relevant info to also serve as a replacement for the printed Certificate as part of e.g. a Digital Product pass.

Closes #929 

<!-- The MS2 and MS3 criteria are intended for merges to the main-branch. For small bug-fixes or during the model development, for instance, when merging to a feature branch, you may decide to not fill out the checklists. However, we recommend to follow the MS2 checklist during the development. The MS3 checklist becomes relevant for merges to the main-branch. -->
## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.11.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] payload names and property identifiers must not contain two consecutive underscores ('__') at any position (e.g. `my__model` is not allowed)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.1.0/datatypes.html) have an example value
- [x] all external / imported models have the state "release"
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

jacewski-bosch committed:
Add Certificate of Analysis 2.0.0 model definition

Makes CatenaXId optional and added disclaimer and documentType properties

jacewski-bosch committed:
Add metadata.json with release status

jacewski-bosch committed:
Update RELEASE_NOTES with version 2.0.0 details

jacewski-bosch committed:
Rename CertificateOfAnalysis to CertificateOfAnalysis.ttl

jacewski-bosch commented on io.catenax.certificate_of_analysis/2.0.0/CertificateOfAnalysis.ttl:
```diff
@@ -0,0 +1,237 @@
+
+:DocumentTypeEntity a samm:Entity ;
+   samm:preferredName "Document type entity"@en ;
+   samm:properties ( ) .
```
something went wrong here. properties missing or remodel as String needed

jacewski-bosch committed:
Update prefixes for CertificateOfAnalysis to version 2.0.0

fixed references to newest versions

jacewski-bosch committed:
Refine descriptions in CertificateOfAnalysis.ttl

Updated descriptions for various characteristics and entities in the Certificate of Analysis ontology to enhance clarity and detail.
changed documentType to Text Characterstic

jacewski-bosch committed:
Fix namespace prefix in CertificateOfAnalysis.ttl

jacewski-bosch committed:
Refactor CertificateOfAnalysis.ttl for zero km failures

added catenaXId for part

jSchuetz88 commented on io.catenax.certificate_of_analysis/2.0.0/CertificateOfAnalysis.ttl:
```diff
@@ -0,0 +1,115 @@
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.zeroKmFailure:1.0.0#> .
```
```suggestion
@prefix : <urn:samm:io.catenax.certificate_of_analysis:2.0.0#> .
```
I assume a copy paste mistake?

jSchuetz88 reviewed (changes requested):
Hi @jacewski-bosch,

thank you for the update. But I'm confused. I assume the name of this aspect model is just a copy paste error (which lead to validation errors)?

Additional first findings:

- [x] Many elements don't have the `samm:description` attribute
- [x] Please use lower case for the `samm:preferredName` attribute

jacewski-bosch committed:
Update io.catenax.certificate_of_analysis/2.0.0/CertificateOfAnalysis.ttl

Co-authored-by: Johann Schütz <147107265+jSchuetz88@users.noreply.github.com>

jacewski-bosch committed:
Add copyright and license details to CertificateOfAnalysis.ttl

Added copyright and licensing information to the file.

jacewski-bosch committed:
Revise Certificate of Analysis model and properties

undoing the last 3 commits that were introduced by mistake

jacewski-bosch commented on io.catenax.certificate_of_analysis/2.0.0/CertificateOfAnalysis.ttl:
```diff
@@ -0,0 +1,138 @@
+   samm:description "This model describes the properties that are necessary to communicate zero kilometer failures of products from the customer to the supplier. Zero km failures are issues with a vehicle component that occur during vehicle assembly."@en ;
+   samm:properties ( :partInformation :vehicle :plantInformation :defect ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
```
How embarrassing... I have manually reverted to the state before the faulty commit. @ther3sa and @jSchuetz88 please proceed with review

jSchuetz88 reviewed (changes requested):
Hi @jacewski-bosch thank you for providing the correct model. 

I mainly have the same finding as for zero_km_failure. Please update the reference `ext-batch` to the shared industry_core model. This should only affect `ext-batch:localIdentifiers`, `ext-batch:manufacturingInformation`and `ext-batch:catenaXId`.

The only impact in content should be the change from `catenaxId` to `globalAssetId`. (Note: This should also be hilighted in the changelog)

jacewski-bosch committed:
Merge branch 'eclipse-tractusx:main' into CertificateOfAnalysis2.0.0

jacewski-bosch committed:
Merge branch 'eclipse-tractusx:main' into CertificateOfAnalysis2.0.0

jacewski-bosch committed:
Merge branch 'eclipse-tractusx:main' into CertificateOfAnalysis2.0.0

jacewski-bosch committed:
add binary document properties in CertificateOfAnalysis.ttl and lower case preferred names

jacewski-bosch committed:
Revise RELEASE_NOTES with detailed field descriptions

Updated the changelog with more detailed descriptions for the added fields and maintained the structure.

jSchuetz88 committed:
Update references to industry_core.common

jSchuetz88 committed:
Format Changelog

jSchuetz88 reviewed (approved):
LGTM, only one optional remark (see above).
